pip/vcs/git.py

```diff
@@ -240,4 +272,19 @@ def update_submodules(self, location):
             cwd=location,
         )
 
+    @classmethod
+    def controls_location(cls, location):
+        if super(Git, cls).controls_location(location):
+            return True
+        try:
+            r = Git().run_command(['rev-parse'],
+                                  cwd=location,
+                                  show_stdout=False,
+                                  raise_on_returncode=False,
+                                  warn_on_returncode=False)
+            return not r
+        except BadCommand:
```

Comment:
Something like that. A similar log is used here: https://github.com/pypa/pip/blob/develop/pip/vcs/__init__.py#L343-L348
But the debug level seems enough for a start :)